esphome/components/max44009/sensor.py

```diff
@@ -0,0 +1,53 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import sensor, i2c
+from esphome.const import (
+    CONF_ID,
+    CONF_MODE,
+    DEVICE_CLASS_ILLUMINANCE,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_LUX,
+)
+
+CODEOWNERS = ["@berfenger"]
+DEPENDENCIES = ["i2c"]
+
+max44009_ns = cg.esphome_ns.namespace("max44009")
+MAX44009Sensor = max44009_ns.class_(
+    "MAX44009Sensor", sensor.Sensor, cg.PollingComponent, i2c.I2CDevice
+)
+
+MAX44009Mode = max44009_ns.enum("MAX44009Mode")
+MODE_OPTIONS = {
+    "auto": MAX44009Mode.MAX44009_MODE_AUTO,
+    "low_power": MAX44009Mode.MAX44009_MODE_LOW_POWER,
+    "continuous": MAX44009Mode.MAX44009_MODE_CONTINUOUS,
+}
+
+CONFIG_SCHEMA = (
+    sensor.sensor_schema(
+        unit_of_measurement=UNIT_LUX,
+        accuracy_decimals=0,
+        device_class=DEVICE_CLASS_ILLUMINANCE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    )
+    .extend(
+        {
+            cv.GenerateID(): cv.declare_id(MAX44009Sensor),
+            cv.Optional(CONF_MODE, default="low_power"): cv.enum(
+                MODE_OPTIONS, upper=False
```

Comment:
```suggestion
                MODE_OPTIONS, lower=True
```

(to also accept `AUTO` for example)